src/overlay/Peer.cpp

```diff
@@ -93,6 +93,34 @@ CapacityTrackedMessage::CapacityTrackedMessage(std::weak_ptr<Peer> peer,
     {
         mMaybeHash = xdrBlake2(msg);
     }
+
+    if (mMsg.type() == TRANSACTION)
+    {
+        auto transaction = TransactionFrameBase::makeTransactionFromWire(
+            self->mAppConnector.getNetworkID(), mMsg.transaction());
+        StellarMessage txMsg;
+        txMsg.type(TRANSACTION);
+        txMsg.transaction() = mMsg.transaction();
+        transaction->getFullHash();
+        transaction->getContentsHash();
+        mTxsMap[*mMaybeHash] = transaction;
```

Comment:
I see that we've been using mes hashes prior to your change, but we don't we just use the tx hash itself for floodgate? I believe it should give use identical filtering capabilites, and would avoid creating a throwaway StellarMessage and doing an extra hashing operation. I think this shouldn't be too difficult implementation wise either, given that the floodmap index is just a generic hash. This would allow us to turn mTxsMap into a set based on TX hash too.

Reply:
That's not very extendable in case StellarMessage format changes in the future. We also need to hash the whole message as opposed to just the transaction part for consistency with other message types references by the floodgate. Otherwise, it's too much of a footgun to have inconsistent hashing semantics. Btw, we switched floodgate and signature verification hashes to BLAKE2 and it's about 2x faster than SHA256 (see Graydon's PR https://github.com/stellar/stellar-core/pull/2795)